geopandas/_vectorized.py

```diff
@@ -0,0 +1,853 @@
+"""
+Compatibility shim for the vectorized geometry operations.
+
+Uses PyGEOS if available/set, otherwise loops through Shapely geometries.
+
+"""
+import warnings
+
+import numpy as np
+
+import shapely.geometry
+import shapely.wkb
+import shapely.wkt
+
+from shapely.geometry.base import BaseGeometry
+
+from . import _compat as compat
+
+try:
+    import pygeos
+except ImportError:
+    geos = None
+
+
+_names = {
+    "NAG": None,
+    "POINT": "Point",
+    "LINESTRING": "LineString",
+    "LINEARRING": "LinearRing",
+    "POLYGON": "Polygon",
+    "MULTIPOINT": "MultiPoint",
+    "MULTILINESTRING": "MultiLineString",
+    "MULTIPOLYGON": "MultiPolygon",
+    "GEOMETRYCOLLECTION": "GeometryCollection",
+}
+
+if compat.USE_PYGEOS:
+    type_mapping = {p.value: _names[p.name] for p in pygeos.GeometryType}
+else:
+    type_mapping = None
+
+
+def _isna(value):
+    """
+    Check if scalar value is NA-like (None or np.nan).
```

Comment:
What about `pd.NA`? That gives False now.

Reply:
> What about pd.NA? That gives False now.

We should indeed start adding support for that, but let's leave that for a separate issue

Reply:
Opened https://github.com/geopandas/geopandas/issues/1335 for this